app/views/posts/_form.html.erb

```diff
@@ -15,7 +15,7 @@
   <%= f.input :body %>
 
   <p>Please select at least two categories.</p>
-  <%= f.association :categories, as: :check_boxes %>
+  <%= f.input :category_ids, as: :check_boxes, collection: Category.all %>
 
```

Comment:
It only invoke `model.categories` that it grabs using `reflect_on_association`. See https://github.com/plataformatec/simple_form/blob/eac1e0108e6c95e291c2065e7005e0cbcb4c2bac/lib/simple_form/form_builder.rb#L554 and https://github.com/plataformatec/simple_form/blob/eac1e0108e6c95e291c2065e7005e0cbcb4c2bac/lib/simple_form/form_builder.rb#L456